exchanges/binanceus/binanceus_test.go

```diff
@@ -0,0 +1,1735 @@
+		t.Error("Binanceus GetAggregateTrades() error", err)
+	}
+}
+
+func TestGetOrderBookDepth(t *testing.T) {
+	t.Parallel()
+	_, er := bi.GetOrderBookDepth(context.TODO(), &OrderBookDataRequestParams{
+		Symbol: currency.NewPair(currency.BTC, currency.USDT),
+		Limit:  1000,
+	})
+	if er != nil {
+		t.Error("Binanceus GetOrderBook() error", er)
+	}
+}
+
+func TestGetCandlestickData(t *testing.T) {
+	t.Parallel()
+	_, er := bi.GetSpotKline(context.Background(), &KlinesRequestParams{
+		Symbol:    currency.NewPair(currency.BTC, currency.USDT),
+		Interval:  kline.FiveMin.Short(),
+		Limit:     24,
+		StartTime: time.Unix(1577836800, 0),
+		EndTime:   time.Unix(1580515200, 0),
+	})
+	if er != nil {
+		t.Error("Binanceus GetSpotKline() error", er)
+	}
+}
+
+func TestGetPriceDatas(t *testing.T) {
+	t.Parallel()
+	_, er := bi.GetPriceDatas(context.Background())
+	if er != nil {
+		t.Error("Binanceus GetPriceDatas() error", er)
+	}
+}
+
+func TestGetSinglePriceData(t *testing.T) {
+	t.Parallel()
+	_, er := bi.GetSinglePriceData(context.Background(), currency.Pair{
+		Base:  currency.BTC,
+		Quote: currency.USDT,
+	})
+	if er != nil {
+		t.Error("Binanceus GetSinglePriceData() error", er)
+	}
+}
+
+func TestGetAveragePrice(t *testing.T) {
+	t.Parallel()
+	_, err := bi.GetAveragePrice(context.Background(), currency.NewPair(currency.BTC, currency.USDT))
+	if err != nil {
+		t.Error("Binance GetAveragePrice() error", err)
+	}
+}
+
+func TestGetBestPrice(t *testing.T) {
+	t.Parallel()
+	_, err := bi.GetBestPrice(context.Background(), currency.NewPair(currency.BTC, currency.USDT))
+	if err != nil {
+		t.Error("Binanceus GetBestPrice() error", err)
+	}
+}
+
+func TestGetPriceChangeStats(t *testing.T) {
+	t.Parallel()
+	_, err := bi.GetPriceChangeStats(context.Background(), currency.NewPair(currency.BTC, currency.USDT))
+	if err != nil {
+		t.Error("Binance GetPriceChangeStats() error", err)
+	}
+}
+
+func TestGetTickers(t *testing.T) {
+	t.Parallel()
+	_, err := bi.GetTickers(context.Background())
+	if err != nil {
+		t.Error("Binance TestGetTickers error", err)
+	}
+}
+
+func TestGetAccount(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	_, er := bi.GetAccount(context.Background())
+	if er != nil {
+		t.Error("Binanceus GetAccount() error", er)
+	}
+}
+
+func TestGetUserAccountStatus(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	_, er := bi.GetUserAccountStatus(context.Background(), 3000)
+	if er != nil {
+		t.Error("Binanceus GetUserAccountStatus() error", er)
+	}
+}
+
+func TestGetUserAPITradingStatus(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	_, er := bi.GetUserAPITradingStatus(context.Background(), 3000)
+	if er != nil {
+		t.Error("Binanceus GetUserAPITradingStatus() error", er)
+	}
+}
+func TestGetTradeFee(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	_, er := bi.GetTradeFee(context.Background(), 3000, "BTC-USDT")
```

Comment:
This one is strange I am still not getting any trade fee from this endpoint and it doesn't seem to error out if the currency pair payload is not even supported by the exchange.